tools/zulint/lister.py

```diff
@@ -26,6 +26,9 @@ def get_ftype(fpath, use_shebang):
             if re.search(r'^#!.*\bpython', first_line):
                 return 'py'
             elif re.search(r'^#!.*sh', first_line):
+                second_line = f.readline()
+                if re.search(r'-\*-(?:.*;)? *mode: *python;.*-\*-', second_line):
+                    return 'py'
```

Comment:
Hmm, since this is shared code (part of `zulint` itself), I suppose if we're going this route we should figure out a more generic way to do this check?

Reply:
Yeah, well, a hard coded list matching `#!.*\bpython` ↦ `py` and four other interpreters and one other “interpreter” was never really generic to begin with, so while the API should be improved I don’t see this as a blocker.

Reply:
Well, one doesn't have to use the `use_shebang` option.  But yeah, I suppose we can just include this in `zulint` with a comment saying it's not generic.